Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1166,14 +1238,56 @@ void MediaPlayerPrivateGStreamer::notifyPlayerOfAudio()
         g_signal_emit_by_name(m_pipeline.get(), "get-audio-pad", i, &pad.outPtr(), nullptr);
         ASSERT(pad);
 
-        String streamId = "A" + String::number(i);
+        auto probeData = adoptRef(*new StreamStartProbeData(this, static_cast<int>(i)));
+        probeData->lock.lock();
+
+        // Since stream-start is not always available at this point, we attach
+        // a probe that can update stream ID once it arrives on this pad.
+        probeData->ref();
+        auto probe = gst_pad_add_probe(
+            pad.get(),
+            (GstPadProbeType) (GST_PAD_PROBE_TYPE_EVENT_BOTH | GST_PAD_PROBE_TYPE_PUSH),
+            streamStartProbeCallback,
+            &probeData.get(),
+            [](gpointer data) {
+                ((StreamStartProbeData *) data)->deref();
+            }
+        );
+        if (!probe) {
+            probeData->deref();
+            GST_WARNING_OBJECT(pipeline(),
+                "Failed to attach stream-start probe, audio track %u will use dummy ID", i);
+        }
+
+        // Check for stream-start *after* attaching the probe,
+        // to avoid creating a race condition where stream-start would arrive
+        // between checking here and creating the probe.
+        String streamId = nullString();
+
+        GstEvent* streamStart = gst_pad_get_sticky_event(pad.get(), GST_EVENT_STREAM_START, 0);
+        if (streamStart) {
+            streamId = audioTrackIdFromStreamStart(streamStart);
+            if (probe)
+                gst_pad_remove_probe(pad.get(), probe);
```

Comment:
If you're running this code it means the probe did not run yet or it is locked at the beginning, but you locked first so you can safely do things. The correct thing here would be removing the probe (no need to check if it is valid because that is always going to be valid, nobody touched that variable) and `probeData->wasProbeRemoved = true` so that if the probe callback was already running but it is locked in the lock, if will find, first thing, that modified and can properly bail out as I told you above.